src/api.rs

```diff
@@ -244,7 +244,7 @@ pub mod request {
         ReadDirFirst:
           - location: Location
           - dir: PathBuf
-          - not_before_filename: Option<PathBuf>
+          - not_before: Option<(PathBuf, bool)>
```

Comment:
I don’t like this tuple, it’s very hard to understand.  What do you think about moving the `bool` into a separate field that only has an effect if `not_before_filename` is set?  Alternatively, a helper struct could be used.

Reply:
I replaced it with an enum with 3 states.